### PR TITLE
Allow IP address to override host name in BigtableSession.

### PR DESCRIPTION
**Pull request by garye**
Resolve host name to IP address for dataflow jobs to work around gRPC issue.

### CODE REVIEW

File: bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSession.java [hunk 1]
```diff
@@ -74,4 +74,6 @@ public void testNoUserAgentBigtableOptions() throws IOException {
     createSession(PROJECT_ID, ZONE_ID, CLUSTER_ID, null);
   }
 
+
```
Comment (sduskis) [inline bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableSession.java]: nit: please remove the blank lines.

**Comment by sduskis** [comment]
LGTM, except for the nit.

**Comment by mgarolera** [comment]
Do we need IP override?
Having a configuration parameter to in BigtableSession.java do
      builder.forAddress(new InetSocketAddress(InetAddress.getByName(host), options.getPort()))
instead of 
      builder.forAddress(host, options.getPort());

should be enough and be more hermetic.
I was thinking about having a "DoHostResolutionAtStartup" parameter.

**Comment by garye** [comment]
I think ideally we wouldn't want to force anyone to add a new parameter to their config since this is a temporary workaround that we will be removing when possible.

**Comment by garye** [comment]
Point taken - it's potentially a bit invasive to solve a particular problem. As per offline discussion let's go with this so we can move forward with a snapshot release and circle back to see if the overrides will prove to be generally useful once the automatic override for dataflow goes away.

**Pull request merged**